libwarpaffine/sliceswriter/SlicesWriterTbb.cpp

```diff
@@ -40,6 +40,17 @@ void CziSlicesWriterTbb::AddSlice(const AddSliceInfo& add_slice_info)
     ++this->number_of_slicewrite_operations_in_flight_;
 }
 
+void CziSlicesWriterTbb::AddAttachment(const std::shared_ptr<libCZI::IAttachment>& attachment)
+{
+    AddAttachmentInfo ai = AddAttachmentInfo();
+    size_t size = 0;
+    ai.ptrData = attachment->GetRawData(&size).get();
+    ai.dataSize = size;
+    ai.SetContentFileType(attachment->GetAttachmentInfo().contentFileType);
+    ai.contentGuid = attachment->GetAttachmentInfo().contentGuid;
+    this->writer_->SyncAddAttachment(ai);
```

Comment:
* The attachment name seems not to be set.
* Line 47 is imho a bit problematic or fishy - 'GetRawData' returns a shared_ptr to some memory block, and I'd say one should consider the raw pointer (returned by 'get()') only to be valid as long as this shared_ptr lives. And, in our case, the lifetime of the shared_ptr ends after the assignment. So, by the time we call into 'SyncAddAttachment' the raw pointer should be considered invalid. Yes, it works as is (at least in current version of libCZI), but I'd argue that the assumption that the pointer remains to be valid is not guaranteed.

My proposal:
```suggestion
    AddAttachmentInfo add_attachment_info;
    const auto& attachment_info = attachment->GetAttachmentInfo();
    add_attachment_info.contentGuid = attachment_info.contentGuid;
    add_attachment_info.SetContentFileType(attachment_info.contentFileType);
    add_attachment_info.SetName(attachment_info.name.c_str());

    size_t size_of_data;
    auto raw_data = attachment->GetRawData(&size_of_data);
    add_attachment_info.ptrData = raw_data.get();
    add_attachment_info.dataSize = static_cast<uint32_t>(size_of_data);

    this->writer_->SyncAddAttachment(add_attachment_info);
```